src/Engine/GameInstance.java

```diff
@@ -70,6 +71,9 @@ public GameInstance(MapMaster map, Weathers weather, GameScenario scenario, bool
     playerCursors = new HashMap<Integer, XYCoord>();
     for( int i = 0; i < commanders.length; ++i )
     {
+      // This is hacky, but hey
+      commanders[i].addUnitModifier(new SandstormModifier());
```

Comment:
sigh

Reply:
The list of `UnitModel` initializations lived in `Commander`'s constructor for the longest time, lol

I'm not sure where I want this sort of thing to go.
Should `GameInstance` (or perhaps `GameRules`?) have a list of modifiers that pertain to all units in the game?
Maybe `GameRules` should have a static function to provide all global modifiers? I imagine we want something like that somewhere, but I'm not sure where it goes.
I also want to allow `UnitModelScheme`s to provide modifiers, though I'm not sure whether I just want to say "put it on the `UnitModel` if it applies to it" and not worry about handling the global-modifier case separately.

Reply:
I mean yeah. It's fine there, at least for now. Just... sigh.

Though frankly, due to weather being tile specific, this is probably as good an approach as any (until a better one appears). Otherwise this restriction would probably just have to be baked into the engine wherever weapon ranges are determined, but fortunately we can filter modifiers on a per-unit basis according to location.